ably/proto/types.go

```diff
@@ -24,3 +31,45 @@ type Data struct {
 	BinaryData []byte  `json:"binaryData" codec:"binaryData"`
 	CipherData []byte  `json:"cipherData" codec:"cipherData"`
 }
+
+// DurationFromMsecs is a time.Duration that is marshaled as a JSON whole number
+// of milliseconds.
+type DurationFromMsecs time.Duration
+
+var _ interface {
+	json.Marshaler
+	json.Unmarshaler
+	codec.Selfer
+} = (*DurationFromMsecs)(nil)
+
+func (t DurationFromMsecs) asMsecs() int64 {
+	return int64(time.Duration(t) / time.Millisecond)
```

Comment:
lets use Duration method provided by stdlib

```suggestion
	return time.Duration(t).Milliseconds()
```